### ISSUE TITLE
How does horovod assign training data when the data is organized by python generator form

### ISSUE THREAD

**Issue by yianzhongguo** [horovod/horovod#2024]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet)
2. Framework version:
3. Horovod version:
4. MPI version:
5. CUDA version:
6. NCCL version:
7. Python version:
8. OS and version:
9. GCC version:

**Checklist:**
1. Did you search issues to find if somebody asked this question before?
2. If your question is about hang, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/running.rst)?
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)?
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)?

**Your question:**
Please ask your question here
Now, I have 4 gpus to train my model. My training data is organized into a python generator which generates a batch step by step and in sequence. When training, since the 4 gpus work synchronously, they will get the same batch of data.  That is that the workers will get the same data, which in fact obeys the principle of parallel training. In this situation there will be no scale for batch size. The benefit of parallel training will not be got. I do not know if I am right. Thank you!

**Comment by tgaddair** [comment]
Hey @yianzhongguo, yes, if you were to do it that way, there would be no benefit to distributed training.  What you would probably prefer to do is partition the data by worker before feeding it to the generator.  That way, each worker only processes a subset of the data each epoch, and you can take advantage of parallel training.

**Comment by yianzhongguo** [comment]
@tgaddair Yes, you are right and I agree with you. But I'm new to distributed training and I do not know how to partition the data by worker before feeding it to the generator, could you detail it, thank you very much!
